test/pagy/countless_test.rb

```diff
@@ -0,0 +1,84 @@
+require_relative '../test_helper'
+require 'pagy/extras/countless'
+
+SingleCov.covered!
+
+describe Pagy::Countless do
+
+  let(:backend) { TestController.new } # page = 3, items = 20
+
+  describe "#finalize" do
+
+    before do
+      @empty_collection = TestCollection.new([])
+      @collection = TestCollection.new(Array(1..59))
+    end
+
+    let(:last_page) { 3 }
+
+    it 'jump to 2 page with empty collection' do
+      proc { Pagy::Countless.new(page: 2).finalize(0) }.must_raise Pagy::OverflowError
+    end
+
+    it 'first page on empty collection' do
+      pagy, _ = backend.send(:pagy_countless, @empty_collection, page: 1)
+
+      pagy.items.must_equal 20
```

Comment:
Please take a look on `pagy.items`. It equals 20 on empty collection. And on last page if page is not full, value can be less the 20. For example `pagy.items.must_equal 4` in one test below. Following this logic, for empty collection items should be 0.

Reply:
That is known and is also documented AFAIK. Quite a long story :).

While the `vars[:items]` remains unchanged, the `@items` is adjusted (for the last non-empty page). The adjustment is needed in order to support the pagination of pre-offsetted and pre-limited collections (let's call them "pO/pL" collections). 

But adjusting the items to 0 for a collection count of 0, would use a limit of 0 in the query, and that would cause a query error with certain DBs. Since adjusting the limit is needed only to retrieve the right items in a pO/pL collection, but when there are 0 items the limit doesn't matter, so it is left unchanged.

Reply:
Ok, I read the documentation, but forgot about it.

Reply:
Now that I think it better... it's probably "documented" only by saying something like "for not-empty pages" or "non-empty collection", but there is not a full explanation of the reason, as in the previous comment. Maybe I should add that somewhere.